static/js/keydown_util.ts

```diff
@@ -31,3 +31,11 @@ export function handle(opts: {
         }
     });
 }
+
+export function is_enter_event(event: any): boolean {
+    // We can have "Enter" events while using an IME
+    // as described in #22062. This check ensures that
+    // we don't trigger our Enter behaviour on those
+    // composing events.
+    return event.which === 13;
+}
```

Comment:
Do we want to check `event.key` also? I don't have a great sense for how this is supposed to work.

Another note -- this PR might read better with a first commit that just extracts `is_enter_event` (with the implementation being still `e.key`) and then a commit that changes the behavior.

Reply:
> Another note -- this PR might read better with a first commit that just extracts `is_enter_event` (with the implementation being still `e.key`) and then a commit that changes the behavior.

Agreed. 👍

I'd be interested in trying the [original proposal](https://chat.zulip.org/#narrow/stream/9-issues/topic/enter.20to.20send.20CJK.20IME.20error.20.2322062/near/1382010):

> Can we make use of `.isComposing` on `KeyboardEvent`s? https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/isComposing

> That should tell us if a `KeyboardEvent` was emitted as part of an [IME composing session](https://developer.mozilla.org/en-US/docs/Glossary/Input_method_editor). When that's the case, we shouldn't let Enter send the Zulip message, right?

@aero31aero, I understand from [CZO](https://chat.zulip.org/#narrow/stream/9-issues/topic/enter.20to.20send.20CJK.20IME.20error.20.2322062/near/1427471) that you don't have a Mac available for testing, so I'll follow through with my offer in May:

> I'm hoping someone who's more familiar with the web front-end code than I am can look at zulip/zulip@bf179b7d2. If that person doesn't have a Mac or otherwise has difficulty reproducing the issue, I can help with manual testing.

I plan to test both the current implementation and one that uses `isComposing`.